src/_css/main.css

```diff
@@ -325,9 +325,13 @@ figure {
   padding: 0;
   text-align: center;
 }
-figure img {
+figure img,
+figure video {
   display: block;
   margin: 0 auto;
+  width: 100%;
```

Comment:
I'm confused. On wide viewports, we wouldn't want images that are smaller than the available width to take up the entire width, would we?

Reply:
If we don't want this, then we have the problem that it uses `height: auto`, but the `width` attribute, which breaks the aspect ratio calculation again:

<img width="188" alt="Screen Shot 2020-04-07 at 12 45 11" src="https://user-images.githubusercontent.com/145676/78660628-b5af1300-78cd-11ea-9187-564425dae620.png">

In practice I think it doesn't matter, since the `body` is capped at `860px`. One way around it would be to also cap `figure` as the container to a maximum width, say `500px`.

Reply:
> One way around it would be to also cap figure as the container to a maximum width, say 500px

This sounds the most promising so far. Are there any downsides to doing that? (aside from hardcoded number)

Reply:
> In practice I think it doesn't matter, since the body is capped at `860px`.

Now I'm even more confused :) Just to be sure, what I meant was that e.g. the image here https://v8.dev/blog/code-caching-for-devs#don%E2%80%99t-change-code gets stretched to the entire width when you give it `width: 100%`, as you'd expect. This is undesirable.

Reply:
```html
<style>
  body { max-width: 860px;}
  figure { max-width: 400px; } /* container */
  figure > img, figure > video {
    width: 100%;
    max-width: 100%;
    height: auto;
  }
</style>

<figure>
  <!-- The container rule would make sure the image is capped at 400px -->
  <img width=1600 height=400 src=foo>
</figure>
```

Reply:
That seems suboptimal; ideally we'd cap images at their actual size, which is what we currently do.

Reply:
The only way that I'm aware of to make this happen while keeping the aspect ratio calculation *during* the loading phase effective would be to add a manual `max-width` with the individual pixel width to each image.

Reply:
> That seems suboptimal; ideally we'd cap images at their actual size, which is what we currently do.

Don't we currently cap them at the container (page) width as well?

Reply:
> Don't we currently cap them at the container (page) width as well?

This is what I was saying [above](https://github.com/v8/v8.dev/pull/376#discussion_r404715511).

Reply:
> > That seems suboptimal; ideally we'd cap images at their actual size, which is what we currently do.
> 
> Don't we currently cap them at the container (page) width as well?

Well, yes, I was talking about the case where images are smaller (not larger) than the container, e.g. on a wide viewport. What I'm really saying is: IMHO we shouldn't stretch images, and hardcoding a number like that `400px` is always gonna result in images smaller than that getting stretched.

Reply:
Are there many non-vector images?

Reply:
> What I'm really saying is: IMHO we shouldn't stretch images, and hardcoding a number like that 400px is always gonna result in images smaller than that getting stretched.

Ah I misunderstood what that code does. Makes sense. Given that we specify dimensions on all images, can we use CSS `min()` + `attr()` perhaps?

> Are there many non-vector images?

There are some screenshots and such, yeah.

Reply:
> Given that we specify dimensions on all images, can we use CSS min() + attr() perhaps?

[Not yet](https://drafts.csswg.org/css-values-4/#attr-notation) 😔

> There are some screenshots and such, yeah.

We could work with a selector that applies the `width: 100%` rule to only SVGs: `img[src$=".svg"]`.

Reply:
> We could work with a selector that applies the width: 100% rule to only SVGs: `img[src$=".svg"]`.

There's SVGs that we don't want to upscale though, e.g. https://v8.dev/blog/liftoff
(Search `dist/` for `width=2**`, `width=3**`, etc. to find more examples.

Reply:
Fair. How about having the best of both worlds? In a pre-processing step before deployment you can for each image with a width ≤ 300px (or whatever threshold you come up with) apply an inline style. This makes sure "wide enough" images get 100% width and auto height, and the smaller images are capped by the inline style. Both will get you the aspect-ratio-dependent space pre-reservation in rendering.

<img width="1586" alt="Screen Shot 2020-04-08 at 10 19 37" src="https://user-images.githubusercontent.com/145676/78761234-a0e08700-7982-11ea-9ff3-6f1b037b7f83.png">

Reply:
@tomayac Wouldn't that break on viewports narrower than whatever the inline `width` style says?

I was pointing out extreme examples before, but I still think we don't want to stretch images in _any_ case (not even when they're SVG and large enough).

Taking a step back, while I appreciate the creative brainstorming here, I think we might be over-engineering this. Responsive images really shouldn't be this hard. Maybe we should go back to https://github.com/v8/v8.dev/issues/370#issuecomment-609744033 and consider that this might be a Chrome bug? @RReverser do you want to file a crbug with your initial findings?

Reply:
> @tomayac Wouldn't that break on viewports narrower than whatever the inline width style says?

True, we'd need to select the threshold accordingly (maybe aimed at KaiOS as the smallest).

> Maybe we should go back to #370 (comment) and consider that this might be a Chrome bug? 

I do not think this is a Chrome bug, since we didn't have `width: 100%`: https://youtu.be/4-d_SoCHeWE?t=438.

Reply:
Hmmm, wait. It works fine in Canary (84) and also Firefox Nightly (77). Could it be that this simply hasn't landed (correctly) in Beta (81)?

Reply:
Yuck, this really bugged me so I dug deeper. Turns out we're far less advanced in the implementation than I thought: https://crbug.com/1045668. It *will* eventually be good, but still isn't.